### PR TITLE
reorg docs nav + add cloud pages

### PR DESCRIPTION
**Pull request by susanev**
## Description
This pr...

- reorgs the docs navigation
- restyles the docs navigation
- introduces 4 new cloud pages
- adds a new partial layout for these 4 new cloud pages

[ux designs in figma](https://www.figma.com/file/CPxzPFMWrtbZOjZXy88418/pulumi-docs?type=design&node-id=661-8147)

## Review deets

- if you give the files changed tab a few minutes to load, it works better
- i kind of hacked the gradient underline for the h1s; my approach will not work for all h1s, so it will need to be reconsidered as part of the type work
- i do want to breakout smaller partials, but will pick that up with later work
---
@anita-trimbur this is ready for a ux design review
things id super appreciate you reviewing closely...

- all new cloud pages
- docs nav interactions
- i think you mentioned not using arrows in certain situations, but i took them from the designs; so pls do let me know if you want any icon updates
- i wasn't sure on the accordion hover styles, let me know if you want any changes on them
- pls interact on a bunch of links, and let me know if you hit any 404s or thingz
----

@kimberleyamackenzie these are the main pages to review; def cool to look at other pages, but don't feel like you need to, since this is a massive pr

- themes/default/layouts/partials/docs/cloud-overview.html
- themes/default/theme/src/scss/docs/_cloud-overview.scss
- themes/default/layouts/partials/docs/toc.html
- themes/default/theme/src/scss/_toc.scss

-----
@sean1588 these are the main pages to review; def cool to look at other pages, but don't feel like you need to, since this is a massive pr

- config/_default/menus.yml
- themes/default/layouts/_default/section.html
- themes/default/layouts/docs/list.html
- themes/default/layouts/partials/docs/cloud-overview.html
- themes/default/theme/src/scss/docs/_cloud-overview.scss
- themes/default/layouts/partials/docs/toc.html
- themes/default/theme/src/scss/_toc.scss
- and Sean i also need ur review on these two prs:
    - https://github.com/pulumi/docs/pull/9061 (nevermind for now on this one, imma leave the SDK files alone for now)
    - https://github.com/pulumi/registry/pull/2479

____

fixes https://github.com/pulumi/docs/issues/8997
fixes https://github.com/pulumi/pulumi-hugo/issues/947
fixes https://github.com/pulumi/pulumi-hugo/issues/2610
fixes https://github.com/pulumi/pulumi-hugo/issues/1790
fixes https://github.com/pulumi/pulumi-hugo/issues/785
fixes https://github.com/pulumi/pulumi-hugo/issues/1500
fixes https://github.com/pulumi/pulumi-hugo/issues/927
fixes https://github.com/pulumi/registry/issues/311

several things need to happen before this can ship.

- [x] get approval from Joe
- [x] get approval from Luke
- [x] where should insights and deployments live? (this was answered today in business review: they will live in pulumi cloud)
- [x] make top headers with children have carets
- [x] make top heading carets function correctly
- [x] get cli commands to colapse
- [x] add clouds content
- [x] move crosswalk to aws folder; delete guides folder
- [x] figure out if i can delete all this left over misc??
- ~[ ] confirm with Luke that aws index of services should be deleted~
- [x] talk to Luke about these aws pages, which do we want which do we not want
- [x] yaml reference: put with yaml
- [x] move package docs from api ref page to language pages
- [x] update api ref page
- [x] add content for using pulumi, adopting pulumi, and access management landing pages
- [x] cross-link faq pages
- [x] get clouds menu to start open
- [x] get rid of left over stuff we don't need; put redirects on current pages
- [x] style fixes to headings, indentation, and arrows; the entire left nav needs to be styled
- [x] add all the redirects 😧 
- [x] all on page links need to get fixed 😧 
- [x] make sure marketing buttons go to the new get started page
- [x] some links in the glossary seem broken
- [x] image on audit logs is broken
- [x] links here are broken /docs/clouds/kubernetes/kubernetes-crosswalk/playbooks/#overview
- [x] all kube guides links seem to be broken
- [x] review all titles and meta descriptions 😧 
- [x] pulumi deployments meta stuff
- [x] pulumi cloud access management meta stuff
- [x] fix border-left
- [x] add line
- [x] fix space on first item
- [x] add scrolling animation
- [x] figure out what the heck this is https://github.com/pulumi/pulumi-hugo/blob/master/themes/default/theme/src/ts/toc.ts
- [x] figure out what the heck this is https://github.com/pulumi/pulumi-hugo/blob/master/themes/default/layouts/shortcodes/pulumi-command.html
- [x] ask luke questions, then update the last of the redirects: aws-guides, cloudfx
- [x] change dynamic resource provider, to dynamic provider
- [x] fix resource options links
- [x] move migreat to landing to adopting
- [x] move migrate from content to adopting parent
- [x] move compare to, to concepts
- [x] add designs to cloud pages
- [x] figure out available versions
- [x] move pulumi cloud between concepts and using
- [x] h1 styles
- [x] update providers with new design
- [x] convert/policies cards need hover styles
- [x] fix kube converter card
- [x] add in kube operator card
- [x] add preview label to providers in preview
- [x] add in kube cluster management card
- [x] responsive cards
- [x] add additional styles to kube cards
- [x] fix kubernetes extension link
- [x] fix aws/cloudfx links
- [x] run link cheker
- [x] review language pages; is SDK in the best place?
- [x] add azure, google cloud, and kube logos to get started button once @anita-trimbur is able to get them
- [x] content updates from Scott
- [x] add accordions
- [x] clean up cloud pages css
- [x] redirects for cli
- [x] figure out redirect and things for available versions
- [x] changes in docs for cli & available versions
- [x] clean up toc html
- [x] clean up toc styles
- [ ] figure out sdk links and redirects
- [ ] do qa!

links to fix
- [x] docs/concepts/function-serialization
- [x] docs/concepts/assets-archives
- [x] /docs/reference

related prs that need to go live at the same time
https://github.com/pulumi/registry/pull/2479
https://github.com/pulumi/docs/pull/9061

## Checklist:

- [x] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [x] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [x] I added aliases (i.e., redirects) for all filename changes.

### CODE REVIEW

File: themes/default/content/templates/container-service/azure/index.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ $ pulumi up
 
 ## Tidying up
 
-You can cleanly destroy the stack and all of its infrastructure with [`pulumi destroy`](/docs/reference/cli/pulumi_destroy):
+You can cleanly destroy the stack and all of its infrastructure with [`pulumi destroy`](/docs/cli/
```
Comment (susanev) [inline themes/default/content/templates/container-service/azure/index.md:90]: fix

File: themes/default/layouts/page/virtual-backgrounds.html [hunk 2]
```diff
@@ -35,15 +35,15 @@ <h2>Get Started with Pulumi</h2>
                 <a
                     id="get-started-aws"
                     class="btn bg-white border border-gray-300 hover:bg-white hover:border-gray-400 p-5 mb-5 flex justify-center w-1/2 mr-4"
-                    href="{{ relref . "/docs/get-started/aws" }}"
+                    href="{{ relref . "/docs/clouds/aws/get-started/" }}"
```
Comment (susanev) [inline themes/default/layouts/page/virtual-backgrounds.html:38]: fix

**Comment by anita-trimbur** [comment]
so so excited this will be shipping soon!! 🚀🚀 just a few style things from me

**cloud pages**
* for the hover styles on the inner cards, can we also change the heading text color to gray900?
* for aws, the how to guide for lambda and serverless goes to the IAM 
* just confirming, do azure and gcp not have a guides card?
* this is definitely my bad, i think i gave you the azure icon to use on the k8s cluster management card - we should probably align to either using the microsoft logo or the azure logo across the cloud pages

**left nav**
* this is totally not blocking just curious, was the multiple left borders for children's children in the nav not feasible? i can revisit styling for the other left nav work

**Review by anita-trimbur — APPROVED** [review]

**Comment by kimberleyamackenzie** [comment]
No matter what I did, I can't load the files tab (which means I can't comment these items inline), so I compared the diff locally and here are some of my thoughts.  Sorry I can't put them inline:

cloud-overview.html
- [ ] line 2 - I don't love inline styles, is there a reason we are prevented from doing this in css?  If there is, then this shouldn't be blocking or anything, but we should really file an issue to take these out of inline styles in the future.
- [ ] line 64 + 66 and some other spots - I would rename these classes to reference the content vs the styling, I think it will be confusing with tailwind classes.  In general I tend to avoid class names that reference the position or styling of the element since they can get moved and then become very confusing.  (exceptions include things like `bold` or `gradient` since those are pretty core to the element itself on the page, it's not a super hard and fast rule).
- [ ] tiny style thing - I'm a little thrown off by the text being gray on rest while the icon is still black here
Screenshot 2023-05-15 at 11 25 37 AM


cloud-overview.scss
- [ ] For the handful of values we user over and over again (for example `#131314` for color) I would make a file level variable like `$text-color: #131314` and use that, so that it's clear all those usages should remain aligned with the same color

toc.html + scss

- [ ] I noticed this when I looked at the most recent preview where the carets should be, i almost wonder if we should consider building those w CSS since they are so fundamental to using the nav that if FA doesn't load, it's very jarring
Screenshot 2023-05-15 at 11 34 43 AM
- [ ] We currently have some smart logic that scrolled the nav down to the current page header item if it was below the fold - we don't have it for sub-items, but with the new nav cleaning up/hiding a lot of sub items, I think it would be nice to add there if we can as a future optimization.


Overall nothing I think should be blocking on these pages, this is so dope and looks so good, y'all have crushed this!!!

**Comment by anita-trimbur** [comment]
small thingz
- [ ] per our slack convo, i think it's totally cool if the inner card icons are gray700 inactive and then gray900 on hover to match the text states
- [ ] for the cards that go to external links, did we want them to open in a new tab? also, should we add arrows on inner cards like for k8s cluster management as an external link affordance, or do we think it's okay as is?

**Review by sean1588 — APPROVED** [review]
amazing job @susanev!! nothing blocking from me here. :tada: :rocket: :1st_place_medal:

**Comment by susanev** [comment]
@anita-trimbur for..
>for the cards that go to external links, did we want them to open in a new tab? also, should we add arrows on inner cards like for k8s cluster management as an external link affordance, or do we think it's okay as is?

can you open a follow-up issue instead and put it in the epic, ready column, and milestone. ya can also assign to me. thank you thank you!!

**Pull request merged**